pkg/collector/corechecks/system/disk/diskv2/disk.go

```diff
@@ -0,0 +1,623 @@
+		}
+	}
+	return nil
+}
+
+func (c *Check) collectDiskMetrics(sender sender.Sender) error {
+	iomap, err := DiskIOCounters()
+	if err != nil {
+		log.Warnf("Unable to get disk iocounters: %s", err)
+		return err
+	}
+	for deviceName, ioCounters := range iomap {
+		log.Debugf("Checking iocounters: [device: %s] [ioCounters: %s]", deviceName, ioCounters)
+		tags := []string{}
+		tags = append(tags, fmt.Sprintf("device:%s", deviceName))
+		tags = append(tags, fmt.Sprintf("device_name:%s", deviceName))
+		tags = append(tags, c.getDeviceTags(deviceName)...)
+		label, ok := c.deviceLabels[deviceName]
+		if ok {
+			tags = append(tags, fmt.Sprintf("label:%s", label), fmt.Sprintf("device_label:%s", label))
+		}
+		c.sendDiskMetrics(sender, ioCounters, tags)
+	}
+
+	return nil
+}
+
+func (c *Check) sendPartitionMetrics(sender sender.Sender, usage *gopsutil_disk.UsageStat, tags []string) {
+	// Disk metrics
+	// For legacy reasons,  the standard unit it kB
+	sender.Gauge(fmt.Sprintf(diskMetric, "total"), float64(usage.Total)/1024, "", tags)
+	sender.Gauge(fmt.Sprintf(diskMetric, "used"), float64(usage.Used)/1024, "", tags)
+	sender.Gauge(fmt.Sprintf(diskMetric, "free"), float64(usage.Free)/1024, "", tags)
+	// FIXME(8.x): use percent, a lot more logical than in_use
+	sender.Gauge(fmt.Sprintf(diskMetric, "in_use"), usage.UsedPercent/100, "", tags)
+
+	// Inodes metrics
+	sender.Gauge(fmt.Sprintf(inodeMetric, "total"), float64(usage.InodesTotal), "", tags)
+	sender.Gauge(fmt.Sprintf(inodeMetric, "used"), float64(usage.InodesUsed), "", tags)
+	sender.Gauge(fmt.Sprintf(inodeMetric, "free"), float64(usage.InodesFree), "", tags)
+	// FIXME(8.x): use percent, a lot more logical than in_use
+	sender.Gauge(fmt.Sprintf(inodeMetric, "in_use"), usage.InodesUsedPercent/100, "", tags)
+}
+
+func (c *Check) sendDiskMetrics(sender sender.Sender, ioCounter gopsutil_disk.IOCountersStat, tags []string) {
+	sender.MonotonicCount(fmt.Sprintf(diskMetric, "read_time"), float64(ioCounter.ReadTime), "", tags)
+	sender.MonotonicCount(fmt.Sprintf(diskMetric, "write_time"), float64(ioCounter.WriteTime), "", tags)
+	// FIXME(8.x): These older metrics are kept here for backwards compatibility, but they are wrong: the value is not a percentage
+	sender.Rate(fmt.Sprintf(diskMetric, "read_time_pct"), float64(ioCounter.ReadTime)*100/1000, "", tags)
+	sender.Rate(fmt.Sprintf(diskMetric, "write_time_pct"), float64(ioCounter.WriteTime)*100/1000, "", tags)
+}
+
+func (c *Check) excludePartition(partition gopsutil_disk.PartitionStat) bool {
+	device := partition.Device
+	if device == "" || device == "none" {
+		device = ""
+		if !c.instanceConfig.AllPartitions {
+			return true
+		}
+	}
+	// Hack for NFS secure mounts
+	// Secure mounts might look like this: '/mypath (deleted)', we should
+	// ignore all the bits not part of the mountpoint name. Take also into
+	// account a space might be in the mountpoint.
+	mountpoint := strings.Split(partition.Mountpoint, " ")[0]
```

Comment:
The Python version uses `mount_point.rsplit(' ', 1)[0]`, so it just drops everything after the last space if there is one

A similar version in Go would be the following (disclaimer, it's untested):
```suggestion
	mountpoint := partition.Mountpoint
	parts := strings.Split(partition.Mountpoint, " ")
	if len(parts) > 1 {
		mountpoint = strings.Join(parts[:len(parts)-1], " ")
	}
```

Reply:
Done